lib/kdl/include/kdl/result.h

```diff
@@ -446,6 +545,62 @@ namespace kdl {
                 return visitor();
             }
         }
+        
+        /**
+         * Applies the given function to given result and returns a new result with the result type of
+         * the given function as its value type.
+         *
+         * If the given result is a success, the function is applied and the result of applying the
+         * function is returned wrapped in a result.
+         * If the given result contains an error, that error is returned as is.
```

Comment:
> that error is returned as-is without invoking `f`.

Reply:
I'll make these changes in a later PR: https://github.com/kduske/TrenchBroom/issues/2983#issuecomment-658147940